src/main/kotlin/org/jetbrains/research/testspark/appstarter/TestSparkStarter.kt

```diff
@@ -0,0 +1,219 @@
+import com.intellij.openapi.application.ApplicationStarter
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.DumbService
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.ProjectManager
+import com.intellij.openapi.roots.ProjectFileIndex
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiJavaFile
+import com.intellij.psi.PsiManager
+import kotlinx.serialization.ExperimentalSerializationApi
+import org.jetbrains.research.testspark.bundles.llm.LLMDefaultsBundle
+import org.jetbrains.research.testspark.core.data.JUnitVersion
+import org.jetbrains.research.testspark.core.data.TestGenerationData
+import org.jetbrains.research.testspark.core.monitor.DefaultErrorMonitor
+import org.jetbrains.research.testspark.data.CodeType
+import org.jetbrains.research.testspark.data.FragmentToTestData
+import org.jetbrains.research.testspark.data.ProjectContext
+import org.jetbrains.research.testspark.data.llm.JsonEncoding
+import org.jetbrains.research.testspark.progress.HeadlessProgressIndicator
+import org.jetbrains.research.testspark.services.LLMSettingsService
+import org.jetbrains.research.testspark.services.PluginSettingsService
+import org.jetbrains.research.testspark.tools.TestProcessor
+import org.jetbrains.research.testspark.tools.ToolUtils
+import org.jetbrains.research.testspark.tools.llm.Llm
+import java.io.File
+import java.nio.file.Paths
+import kotlin.system.exitProcess
+
+/**
+ * This class is responsible for generating and running tests based on the provided arguments in headless mode.
+ */
+class TestSparkStarter : ApplicationStarter {
+    @Deprecated("Specify it as `id` for extension definition in a plugin descriptor")
+    override val commandName: String = "testspark"
+
+    /** Sets main (start) thread for IDE in headless as not edt. */
+    override val requiredModality: Int = ApplicationStarter.NOT_IN_EDT
+
+    @Suppress("TooGenericExceptionCaught")
+    @OptIn(ExperimentalSerializationApi::class)
+    override fun main(args: List<String>) {
+        // Project path
+        val projectPath = args[1]
+        // Path to the target file (.java file)
+        val cutSourceFilePath = Paths.get(projectPath, args[2]).toAbsolutePath()
+        // CUT name (<package-name>.<class-name>)
+        val classUnderTestName = args[3]
+        // Paths to compilation output of the project under test (seperated by ':')
+        val projectClassPath = args[4]
+        val classPath = "$projectPath${ToolUtils.sep}$projectClassPath"
+        // JUnit Version
+        val jUnitVersion = args[5]
+        // Selected model
+        val model = args[6]
+        // Token
+        val token = args[7]
+        // Filepath to a file containing the prompt template
+        val promptTemplateFile = args[8]
+        // Output directory
+        val output = args[9]
+
+        println("Test generation requested for $projectPath")
```

Comment:
Where this data will be printed? Why not `log()`?

Reply:
Since we use this for internal assessment, we would like to keep these logs separate from the general test generation procedure. So, I decided to make in simple println for now.